### PR TITLE
Verify BSD+MIT+Go+UPL+ISC license mentions in `NOTICE` file + expose in Nessie server, GC-tool, Admin-Tool and CLI/REPL

### PR DESCRIPTION
**Pull request by snazy**
Dependencies that are not Apache licensed but MIT/BSD/Go/UPL/ISC licensed must include an entry in the `NOTICE` file. This change verifies that an "entry" for those dependencies is present.

This change also allows "retrieving" the `NOTICE` via:
* `/NOTICE.txt` from Nessie/Quarkus server
* via `HELP LICENSE` in the CLI/REPL
* via the new `show-licenses` command in the GC tool

Also:
* Adds a new "global" convenience `codeChecks` task, that includes license-checks, spotless-check and checkstyle.
* Adds MIT/BSD/ISC licenses from the Nessie web UI to the `NOTICE` file

### CODE REVIEW

File: build-logic/build.gradle.kts [hunk 1]
```diff
@@ -33,3 +33,15 @@ dependencies {
 }
 
 tasks.withType<Test>().configureEach { useJUnitPlatform() }
+
+tasks.register("compileAll").configure {
```
Comment (snazy) [inline build-logic/build.gradle.kts:37]: Adding these two tasks here as well, as a convenience

File: build-logic/src/main/kotlin/nessie-common-src.gradle.kts [hunk 2]
```diff
@@ -219,3 +222,17 @@ class MemoizedGitInfo {
     }
   }
 }
+
+afterEvaluate {
+  tasks.register("codeChecks").configure {
```
Comment (snazy) [inline build-logic/src/main/kotlin/nessie-common-src.gradle.kts:227]: Convenience task to run "all the code checks"

File: servers/quarkus-server/build.gradle.kts [hunk 3]
```diff
@@ -51,6 +51,7 @@ dependencies {
   implementation(project(":nessie-rest-common"))
   implementation(project(":nessie-rest-services"))
   implementation(project(":nessie-versioned-spi"))
+  implementation(project(":nessie-notice"))
```
Comment (snazy) [inline servers/quarkus-server/build.gradle.kts:54]: That's all what's needed to make `curl http://127.0.0.1:19120/NOTICE.txt` work

File: build-logic/src/main/kotlin/NoticeReportValidation.kt [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    val missing = mutableMapOf<String, String>()
+
+    data.allDependencies.forEach { mod ->
+      val licenses =
+        (mod.manifests.map { it.license } +
+            mod.licenseFiles.flatMap { it.fileDetails }.map { it.license } +
+            mod.poms.flatMap { it.licenses }.map { it.name })
+          .distinct()
+
+      if (!licenses.any { needsNoNotice(it) } && licenses.any { needsNotice(it) }) {
```
Comment (adutra) [inline build-logic/src/main/kotlin/NoticeReportValidation.kt:54]: I suppose this is meant for dual licensing, maybe add a comment explaining the rationale?

**Review by adutra — APPROVED** [review]

File: build-logic/src/main/kotlin/NoticeReportValidation.kt (hunk 4 above)
Comment (snazy) [reply build-logic/src/main/kotlin/NoticeReportValidation.kt:54]: Correct - it's for (most) dependencies that use more than one license.

**Pull request merged**